src/openms/source/FORMAT/HANDLERS/MzMLHandler.cpp

```diff
@@ -1393,7 +1400,23 @@ namespace OpenMS
         }
       }
 
-      if (unit_accession != "") termValue.setUnit(unit_accession);
+      if (unit_accession != "")
+      {
+        if (unit_accession.hasPrefix("UO:"))
+        {
+          termValue.setUnit(unit_accession.suffix(unit_accession.size() - 3).toInt());
```

Comment:
@timosachsenberg this should be pretty safe, I dont see a case for `MS:xxx` or `UO:xxx` where xxx is not a integer